### PR TITLE
Hindi TN Support for Cardinal, Decimal, Fraction, Date, Time, Money and Measure

### PR DESCRIPTION
**Pull request by ngachchi**
# What does this PR do ?

This PR introduces Hindi support for a wide range of numerical and temporal formats, including:

- **Cardinal numbers:** Natural numbers (e.g., एक, दो, तीन)
- **Decimal numbers:** Numbers with decimal points (e.g., दशमलव दो दशमलव पांच)
- **Fractions:** Rational numbers expressed as ratios (e.g., एक बटा दो)
- **Dates:** Various date formats (e.g., आज, कल, १४ नवंबर २०२४)
- **Time:** Time formats (e.g., दो बजकर पांच मिनट)
- **Money:** Monetary amounts (e.g., दस रुपये पचास पैसे)
- **Measure**: Units of measurement (e.g., दस किलोमीटर)

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Have you signed your commits? Use ``git commit -s`` to sign.
- [x] Do all unittests finish successfully before sending PR?
   1) ``pytest`` or (if your machine does not have GPU) ``pytest --cpu`` from the root folder (given you marked your test cases accordingly `@pytest.mark.run_only_on('CPU')`).
   2) Sparrowhawk tests ``bash tools/text_processing_deployment/export_grammars.sh --MODE=test ...``
- [x] If you are adding a new feature: Have you added test cases for both `pytest` and Sparrowhawk [here](tests/nemo_text_processing).
- [x] Have you added ``__init__.py`` for every folder and subfolder, including `data` folder which has .TSV files?
- [x] Have you followed codeQL results and removed unused variables and imports (report is at the bottom of the PR in github review box) ?
- [x] Have you added the correct license header `Copyright (c) 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.` to all newly added Python files?
- [ ] If you copied [nemo_text_processing/text_normalization/en/graph_utils.py](nemo_text_processing/text_normalization/en/graph_utils.py) your header's second line should be `Copyright 2015 and onwards Google, Inc.`. See an example [here](https://github.com/NVIDIA/NeMo-text-processing/blob/main/nemo_text_processing/text_normalization/en/graph_utils.py#L2).
- [x] Remove import guards (`try import: ... except: ...`) if not already done.
- [ ] If you added a new language or a new feature please update the [NeMo documentation](https://github.com/NVIDIA/NeMo/blob/main/docs/source/nlp/text_normalization/wfst/wfst_text_normalization.rst) (lives in different repo).
- [x] Have you added your language support to [tools/text_processing_deployment/pynini_export.py](tools/text_processing_deployment/pynini_export.py).
  

**PR Type**:
- [x] New Feature
- [ ] Bugfix
- [ ] Documentation
- [ ] Test

If you haven't finished some of the above items you can still open "Draft" PR.

### CODE REVIEW

File: nemo_text_processing/text_normalization/hi/data/whitelist/alternatives.tsv
[the remark this answers is not in the document]
Comment (ngachchi) [reply nemo_text_processing/text_normalization/hi/data/whitelist/alternatives.tsv:1]: Yes, you're correct and I think we can skip the whitelist for now as for Hindi language we don't have much abbreviations in this context like for address, titles, it's just the entire word itself.

**Review by ngachchi — COMMENTED** [review]
remaining files from the whitelist data class will be removed and single would be there

File: tests/nemo_text_processing/hi/test_word.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    def test_norm(self, test_input, expected):
+        pred = self.normalizer.normalize(test_input, verbose=False, punct_post_process=True)
+        assert pred == expected
+
+    # @parameterized.expand(parse_test_case_file('hi/data_inverse_text_normalization/test_cases_word.txt'))
+    # @pytest.mark.run_only_on('CPU')
+    # @pytest.mark.unit
+    # def test_denorm(self, test_input, expected):
+    #     pred = self.inverse_normalizer.inverse_normalize(test_input, verbose=False)
+    #     assert pred.strip() == expected.strip()
```
[the remark this answers is not in the document]
Comment (ngachchi) [reply tests/nemo_text_processing/hi/test_word.py]: removed unused statements

File: tests/nemo_text_processing/hi/test_whitelist.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    def test_norm(self, test_input, expected):
+        pred = self.normalizer.normalize(test_input, verbose=False)
+        assert pred.strip() == expected.strip()
+
+    # @parameterized.expand(parse_test_case_file('hi/data_inverse_text_normalization/test_cases_whitelist.txt'))
+    # @pytest.mark.run_only_on('CPU')
+    # @pytest.mark.unit
+    # def test_denorm(self, test_input, expected):
+    #     pred = self.inverse_normalizer.inverse_normalize(test_input, verbose=False)
+    #     assert pred.strip() == expected.strip()
```
[the remark this answers is not in the document]
Comment (ngachchi) [reply tests/nemo_text_processing/hi/test_whitelist.py]: removed unused statements

File: nemo_text_processing/text_normalization/hi/graph_utils.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+import string
+from pathlib import Path
+from typing import Dict
+
+import pynini
+from pynini import Far
+from pynini.export import export
+from pynini.lib import byte, pynutil, utf8
+
+from nemo_text_processing.text_normalization.hi.utils import get_abs_path, load_labels
```
[the remark this answers is not in the document]
Comment (ngachchi) [reply nemo_text_processing/text_normalization/hi/graph_utils.py]: removed unused imports

File: nemo_text_processing/text_normalization/hi/taggers/word.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import pynini
+from pynini.lib import pynutil
+
+from nemo_text_processing.text_normalization.hi.graph_utils import (
+    MIN_NEG_WEIGHT,
+    NEMO_NOT_SPACE,
+    NEMO_SIGMA,
+    GraphFst,
+    convert_space,
+)
```
[the remark this answers is not in the document]
Comment (ngachchi) [reply nemo_text_processing/text_normalization/hi/taggers/word.py]: removed unused import

**Review by mgrafu — APPROVED** [review]

**Pull request merged**